crates/ide/src/annotations.rs

```diff
@@ -141,3 +143,268 @@ pub(crate) fn resolve_annotation(db: &RootDatabase, mut annotation: Annotation)
 
     annotation
 }
+
+#[cfg(test)]
+mod tests {
+    use ide_db::base_db::{FileId, FileRange};
+    use syntax::{TextRange, TextSize};
+
+    use crate::{fixture, Annotation, AnnotationConfig, AnnotationKind, RunnableKind};
+
+    fn get_annotations(
+        ra_fixture: &str,
+        annotation_config: AnnotationConfig,
+    ) -> (FileId, Vec<Annotation>) {
+        let (analysis, file_id) = fixture::file(ra_fixture);
+
+        let annotations: Vec<Annotation> = analysis
+            .annotations(file_id, annotation_config)
+            .unwrap()
+            .into_iter()
+            .map(move |annotation| analysis.resolve_annotation(annotation).unwrap())
+            .collect();
+
+        if annotations.len() == 0 {
+            panic!("unresolved annotations")
+        }
+
+        (file_id, annotations)
+    }
+
+    macro_rules! check_annotation {
+        ( $ra_fixture:expr, $config:expr, $item_positions:expr, $pattern:pat, $checker:expr ) => {
+            let (file_id, annotations) = get_annotations($ra_fixture, $config);
+
+            annotations.into_iter().for_each(|annotation| {
+                assert!($item_positions.contains(&annotation.range));
+
+                match annotation.kind {
+                    $pattern => $checker(file_id),
+                    _ => panic!("Unexpected annotation kind"),
+                }
+            });
+        };
+    }
```

Comment:
We generally avoid macros in tests, as they make tests harder to understand and modify. Instead, we use expect tests. Tests for runnables are a good example: https://github.com/rust-analyzer/rust-analyzer/blob/11ebbac3a4a962701639bb14a4a5a7309399d7e8/crates/ide/src/runnables.rs#L648-L700

Reply:
Oh, my bad 😅